### ISSUE TITLE
Email Generation has way too many edge cases

### ISSUE THREAD

**Issue by jlink** [jlink/jqwik#133]
## Testing Problem

Email Generation has way too many edge cases which leads to only few generated email addresses to be
just normal and random

## Suggested Solution

Restrict edge cases to just the most valid ones, e.g.
- `a@a.aa`
- `"a"@a.aa`
- `" "@a.aa`
- `a@[1.1.1.1]` or `[0.0.0.0]`?
- `a@[FF::FF:FF:FF:FF:FFFF:FF]`
- etc.

**Comment by jlink** [comment]
@zinki97 Any suggestions about edge cases for ip4 and ip6 addresses?

**Comment by zinki97** [comment]
> Any suggestions about edge cases for ip4 and ip6 addresses?

I think edge cases are 0.0.0.0, 255.255.255.255, 0:0:0:0:0:0:0:0 (or equal ones like 0:0:0::0:0:0:0) and FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF

**Comment by luvarqpp** [comment]
I vote for '::1' (short form of localhost address.)

**Comment by jlink** [comment]
This is the current list of edge cases:

```
A@[::]
A@[0:0:0:0:0:0:0:0]
A@[2:2:2:2:2:2:2:2]
A@[1:1:1:1:1:1:1:1]
A@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
A@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
Z@[::]
Z@[0:0:0:0:0:0:0:0]
Z@[2:2:2:2:2:2:2:2]
Z@[1:1:1:1:1:1:1:1]
Z@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
Z@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
!@[::]
!@[0:0:0:0:0:0:0:0]
!@[2:2:2:2:2:2:2:2]
!@[1:1:1:1:1:1:1:1]
!@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
!@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
"A"@[::]
"A"@[0:0:0:0:0:0:0:0]
"A"@[2:2:2:2:2:2:2:2]
"A"@[1:1:1:1:1:1:1:1]
"A"@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"A"@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
"Z"@[::]
"Z"@[0:0:0:0:0:0:0:0]
"Z"@[2:2:2:2:2:2:2:2]
"Z"@[1:1:1:1:1:1:1:1]
"Z"@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"Z"@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
" "@[::]
" "@[0:0:0:0:0:0:0:0]
" "@[2:2:2:2:2:2:2:2]
" "@[1:1:1:1:1:1:1:1]
" "@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
" "@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
"]"@[::]
"]"@[0:0:0:0:0:0:0:0]
"]"@[2:2:2:2:2:2:2:2]
"]"@[1:1:1:1:1:1:1:1]
"]"@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"]"@[FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE:FFFE]
```

Obviously domain hosts and ip4 addresses are completely missing. More questions I have:
- Is ! really a valid local part?
- Is " " a valid quoted local part?
- Do [2:2:2:2:2:2:2:2] and [1:1:1:1:1:1:1:1] offer additional value?

**Comment by jlink** [comment]
Here's the list of all 63 edge cases being generated in 1.3.9-SNAPSHOT:

```
!@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
a@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
0@[255.255.255.255]
A@[255.255.255.255]
"A"@[0.0.0.0]
!@a.aa
" "@a.aa
a@[0.0.0.0]
"A"@[255.255.255.255]
a@[0:0:0:0:0:0:0:0]
A@[::]
0@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
" "@[::]
A@[0.0.0.0]
!@[::]
"a"@[127.0.0.1]
0@[0:0:0:0:0:0:0:0]
" "@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
" "@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
a@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
"a"@[::]
!@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
" "@[0.0.0.0]
A@[127.0.0.1]
a@[127.0.0.1]
"a"@a.aa
A@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
!@[127.0.0.1]
" "@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"a"@[0:0:0:0:0:0:0:0]
"A"@[127.0.0.1]
"A"@[0:0:0:0:0:0:0:0]
"a"@[255.255.255.255]
0@a.aa
"a"@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
"a"@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
"A"@[::]
a@[255.255.255.255]
a@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
0@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
" "@[127.0.0.1]
a@[::]
0@[0.0.0.0]
"A"@a.aa
A@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"a"@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"A"@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
"a"@[0.0.0.0]
!@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
A@[0:0:0:0:0:0:0:0]
0@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
"A"@a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.a.aa
" "@[0:0:0:0:0:0:0:0]
!@[255.255.255.255]
0@[::]
!@[0.0.0.0]
A@a.aa
!@[0:0:0:0:0:0:0:0]
0@[127.0.0.1]
A@[ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff]
a@a.aa
"A"@[FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF:FFFF]
" "@[255.255.255.255]
```

Anything missing? Anything in that shouldn't be in?

**Comment by jlink** [comment]
Added in https://github.com/jlink/jqwik/commit/c7813b94d5e0be7d64f07c934a6fb5ae955cadb9
Available in 1.3.9-SNAPSHOT
